volatility3/framework/plugins/windows/stack.py

```diff
@@ -0,0 +1,118 @@
+import struct
+
+from typing import List
+
+from volatility3.framework.symbols.windows import extensions
+from volatility3.framework.configuration import requirements
+from volatility3.framework import interfaces, renderers, constants
+from volatility3.framework.renderers import format_hints
+from volatility3.plugins.windows import pslist
+
+
+class Stack(interfaces.plugins.PluginInterface):
+    """Lists the Stack boundaries and dump Stack"""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (3, 0, 1)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows Kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.VersionRequirement(
+                name="pslist", component=pslist.PsList, version=(3, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Process IDs to operate on",
+                optional=False,
+            ),
+            requirements.BooleanRequirement(
+                name="dump",
+                description="Whether to dump the stack",
+                default=False,
+                optional=True,
+            ),
+        ]
+
+    def _generator(self):
+        file_output = "Disabled"
+
+        kernel = self.context.modules[self.config["kernel"]]
+        filter_func = pslist.PsList.create_pid_filter(self.config.get("pid"), None)
+
+        procs: List[extensions.EPROCESS] = pslist.PsList.list_processes(
+            context=self.context,
+            kernel_module_name=self.config["kernel"],
+            filter_func=filter_func,
+        )
+
+        for proc in procs:
+            proc_layer_name = proc.add_process_layer()
+            proc_layer = self.context.layers[proc_layer_name]
+
+            thread_list: List[extensions.ETHREAD] = list(
+                proc.ThreadListHead.to_list(
+                    f"{kernel.symbol_table_name}{constants.BANG}_ETHREAD",
+                    "ThreadListEntry",
+                )
+            )
+
+            # no need to parse the time
+            active_thread_list: List[extensions.ETHREAD] = [
+                t for t in thread_list if t.ExitTime.QuadPart < 0
+            ]
+
+            for thread in active_thread_list:
+                trap_frame = thread.Tcb.TrapFrame.dereference()
+                thread_rsp = trap_frame.Rsp
+
+                # first entry of TEB is NT_TIB
+                # Stack Base is at NT_TIB + 8
+                # https://github.com/wine-mirror/wine/blob/master/include/winternl.h#L494
+                # https://github.com/wine-mirror/wine/blob/master/include/winnt.h#L2445
+                stack_base = struct.unpack(
```

Comment:
Please try to avoid reading bytes directly.  Either make a struct or at least instantiate a new object of the right type for windows on it.  Having to 0 index the results of unpack basically throws away all the extra goodness we provide through the volatility model (keeping track of the type, the offset, which layer it came from, etc).

The entire volatility framework is a giant parser, so if you're resorting to struct.unpack yourself please consider if there's *no* other way of doing it (or if you're doing it so often efficiency comes into question, which ones per thread, isn't the case here).